dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -3555,6 +3557,69 @@ def setUpClass(cls):
                 cls.data, len(cls.data), options=profiler_options_full
             )
 
+    def test_add_profiles_null_count_not_enabled(self):
+        profiler_options_null_count = ProfilerOptions()
+        profiler_options_null_count.set(
+            {
+                "*.is_enabled": False,
+                "row_statistics.*.is_enabled": True,
+                "row_statistics.null_count.is_enabled": True,
+            }
+        )
+        profiler_options_null_disabled = ProfilerOptions()
+        profiler_options_null_disabled.set(
+            {
+                "*.is_enabled": False,
+                "row_statistics.*.is_enabled": True,
+                "row_statistics.null_count.is_enabled": False,
+            }
+        )
+        data = pd.DataFrame([1, None, 3, 4, 5, None, 1])
+        with test_utils.mock_timeit():
+            profiler_w_null_count = dp.StructuredProfiler(
+                data[:2], options=profiler_options_null_count
+            )
+            profiler_w_disabled_null_count = dp.StructuredProfiler(
+                data[2:], options=profiler_options_null_disabled
+            )
+
+        with self.assertRaisesRegex(
+            ValueError,
+            "Attempting to merge two profiles with null row "
+            "count option enabled on one profile but not the other.",
+        ):
+            profiler_w_null_count + profiler_w_disabled_null_count
+
+    def test_adding_profiles_of_mismatched_null_count_options(self):
+        profiler_options_null_count = ProfilerOptions()
+        profiler_options_null_count.set(
+            {
+                "*.is_enabled": False,
+                "row_statistics.*.is_enabled": True,
+                "row_statistics.null_count.is_enabled": True,
+            }
+        )
+        profiler_options_null_disabled = ProfilerOptions()
+        profiler_options_null_disabled.set(
+            {
+                "*.is_enabled": False,
+                "row_statistics.*.is_enabled": True,
+                "row_statistics.null_count.is_enabled": False,
+            }
+        )
+        data = pd.DataFrame([1, None, 3, 4, 5, None, 1])
+        with test_utils.mock_timeit():
+            profiler_w_null_count = dp.StructuredProfiler(
+                data[:2], options=profiler_options_null_count
+            )
+            profiler_w_disabled_null_count = dp.StructuredProfiler(
+                data[2:], options=profiler_options_null_disabled
+            )
+        self.assertLess(0, profiler_w_null_count.row_has_null_count)
+        self.assertLess(0, profiler_w_null_count.row_is_null_count)
+        self.assertEqual(0, profiler_w_disabled_null_count.row_has_null_count)
+        self.assertEqual(0, profiler_w_disabled_null_count.row_is_null_count)
```

Comment:
I think we still need to split this out as this isn't testing mismatching of add in terms of the test function naming.

Reply:
Actually, just noticing the swapping of names above.
If we change this to `test_add_profiles_null_count_not_enabled` we only need to test one profile and it's values.

Reply:
We still need to include the tests in this comment: https://github.com/capitalone/DataProfiler/pull/871/files#r1230135673